sunpy/map/mapbase.py

```diff
@@ -3195,6 +3201,10 @@ def reproject_to(self, target_wcs, *, algorithm='interpolation', return_footprin
         # Create and return a new GenericMap
         outmap = GenericMap(output_array, target_wcs.to_header(),
                             plot_settings=self.plot_settings)
+        if preserve_date:
+            # date-avg is the time used as the coordinate frame obstime and
+            # will already be set by the target_wcs
+            outmap.meta['date-obs'] = self.date.isot
```

Comment:
I think the correct way to do this would be with the `_set_date` method: https://github.com/sunpy/sunpy/blob/228be4348fedfa5e5ecb8c898a18dbf06226e547/sunpy/map/mapbase.py#L1031

Some classes override this to fix the metadata in their way.